### PR TITLE
edge_color

### PR DESCRIPTION
**Pull request by noragak**

### CODE REVIEW

**Comment by ericmjl** [comment]
@noragak thank you for the PR! I will take a look at it this evening (my local time).

File: AUTHORS.rst [hunk 1]
```diff
@@ -10,4 +10,4 @@ Development Lead
 Contributors
 ------------
 
-None yet. Why not be the first?
+* Nora Kassner <nora.kassner@t-online.de>
```
Comment (ericmjl) [inline AUTHORS.rst]: yayyy!!! :smile:

File: nxviz/plots.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def __init__(self, graph, node_order=None, node_size=None,
         # Set graph object
         self.graph = graph
         self.nodes = list(graph.nodes())  # keep track of nodes separately.
-
+        self.edges = list(graph.edges())
```
Comment (ericmjl) [inline nxviz/plots.py]: Please add comment analogous to `# keep track of nodes separately.`

File: nxviz/utils.py [hunk 3]
```diff
@@ -130,4 +131,6 @@ def num_discrete_groups(data_container):
     'Accent_8': qualitative.Accent_8,
     'continuous': sequential.YlGnBu_9,
     'diverging': diverging.RdBu_11,
+    'weights': cubehelix_palette(50, hue=0.05, rot=0, light=0.9, dark=0,
```
Comment (ericmjl) [inline nxviz/utils.py]: Nice! I like the suggested color palette.

File: examples/circos/edge_color.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+"""
+Shows different edge colors. Either categorial/ordinal or continuous
+"""
+import matplotlib.pyplot as plt
+import networkx as nx
+from nxviz.plots import CircosPlot
+
+nodelist = [('a'),
+                ('b'),
```
Comment (ericmjl) [inline examples/circos/edge_color.py]: Indentation probably not necessary here.

File: examples/circos/edge_color.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                ('c'),
+                ('d'),
+                ('e'),
+                ('f')
+                ]
+edgelist1 = [('a', 'b', {'weight': 0.1}),
+            ('a', 'c', {'weight': 0.2}),
```
Comment (ericmjl) [inline examples/circos/edge_color.py]: `pycodestyle` is asking for the parentheses the be lined up. Please make this change.

File: examples/circos/edge_color.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+            ('b', 'd', {'weight': 0.6}),
+            ('c', 'd', {'weight': 0.7}),
+            ('e', 'd', {'weight': 0.8}),
+            ('e', 'f', {'weight': 1.0}),
+            ]
+edgelist2 = [('a', 'b', {'class': 1}),
+            ('a', 'c', {'class': 3}),
```
Comment (ericmjl) [inline examples/circos/edge_color.py]: Same as comment for line 16.

File: examples/circos/edge_color.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+                ('e'),
+                ('f')
+                ]
+edgelist1 = [('a', 'b', {'weight': 0.1}),
+            ('a', 'c', {'weight': 0.2}),
+            ('b', 'd', {'weight': 0.6}),
+            ('c', 'd', {'weight': 0.7}),
+            ('e', 'd', {'weight': 0.8}),
+            ('e', 'f', {'weight': 1.0}),
+            ]
```
Comment (ericmjl) [inline examples/circos/edge_color.py]: `pycodestyle` is also complaining about this one. I think it needs one more space before the closing bracket.

File: examples/circos/edge_color.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+edgelist2 = [('a', 'b', {'class': 1}),
+            ('a', 'c', {'class': 3}),
+            ('b', 'd', {'class': 8}),
+            ('c', 'd', {'class': 10}),
+            ('e', 'd', {'class': 5}),
+            ('e', 'f', {'class': 5}),
+            ]
```
Comment (ericmjl) [inline examples/circos/edge_color.py]: Same as comment for line 21.

File: tests/test_nxviz.py [hunk 9]
```diff
@@ -114,3 +137,20 @@ def test_init_node_colors():
     G = make_graph_for_grouping()  # noqa
     b = BasePlot(graph=G)
     assert len(set(b.node_colors)) == 1
+
+
+def test_init_edge_colors():
+    """
+    Does two checks:
+    1. If edge_color is not passed in as a keyword argument, check that
+       self.edge_colors is a list of 'blue', of length (number of nodes).
```
Comment (ericmjl) [inline tests/test_nxviz.py]: I think the code in `plots.py` states that it should be a list of 'black'. Please change this accordingly.

File: tests/test_nxviz.py [hunk 10]
```diff
@@ -114,3 +137,20 @@ def test_init_node_colors():
...
+    2. If edge_color is passed in as a keyword argument, check that
+       self.node_colors is a list with more than one element.
+    """
+    G = make_graph_for_edges()  # noqa
+    b = BasePlot(graph=G, edge_color="weight")
+    assert len(set(b.edge_colors)) > 1
+
+    G = make_graph_for_grouping()  # noqa
+    b = BasePlot(graph=G)
+    assert len(set(b.edge_colors)) == 1
```
Comment (ericmjl) [inline tests/test_nxviz.py]: Not sure why this one is failing. Do you have any hypotheses as to why?

**Comment by ericmjl** [comment]
@noragak alternatively, if you're busy, I'm happy to do the finishing touches, as long as I'm given permission to edit the PR.

**Comment by noragak** [comment]
Hi, I can do it this evening (German time) after work! Thank you :)

File: nxviz/utils.py (hunk 3 above)
Comment (noragak) [reply nxviz/utils.py]: Yes this way even weights close to 0 are visible! Which is needed for NUPACK plots.

File: tests/test_nxviz.py (hunk 10 above)
Comment (noragak) [reply tests/test_nxviz.py]: Maybe two blank lines are needed. But I am not sure..

Comment (ericmjl) [reply tests/test_nxviz.py]: If you run the test locally, does it pass?

Comment (noragak) [reply tests/test_nxviz.py]: You mean if I can run "python test_plots.py" right? Sorry that is my first PR with someone outside of my group and with GitHub, so excuse me for being a bit slow and maybe misunderstandig things. Thank you for you time :)

Comment (noragak) [reply tests/test_nxviz.py]: But yes when I run test_plot.py there are no complains

Comment (ericmjl) [reply tests/test_nxviz.py]: No worries about that! I was at the place you’re at before :). By running it locally, I meant calling ‘py.test’ In the terminal, or by running the code in the test function manually. Could you try that out to see if things are working locally?

Comment (noragak) [reply tests/test_nxviz.py]: Found the mistake. I am correcting it right now. Whats to to next?

**Comment by ericmjl** [comment]
We’re almost there!! There’s just a few pycodestyle fixes that Travis is complaining about. Could you look at the log to see which lines need to be fixed? https://travis-ci.org/ericmjl/nxviz/builds/344011718?utm_source=github_status&utm_medium=notification

I’m on the train home, it’s a novel experience for me reviewing from my phone!

**Comment by ericmjl** [comment]
If it helps, do install pycodestyle and then run the command “pycodestyle .” from inside nxviz. That may help you catch the errors before pushing.

**Comment by noragak** [comment]
Now it passed :)

**Comment by ericmjl** [comment]
Woohoo!!! All tests pass! I’m going to merge it in. Thanks @noragak! Do you have a Twitter handle? I’d love to tweet about it :)

**Comment by noragak** [comment]
Perfect! Thank you for your time. And yes I do have twitter since a couple of days actually :)

**Comment by noragak** [comment]
https://twitter.com/KassnerNora

**Pull request merged**